packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2031,6 +1925,130 @@ export class ContainerRuntime
 		await this.garbageCollector.initializeBaseState();
 	}
 
+	private async initializeSummarizer(loader: ILoader): Promise<void> {
+		const { maxOpsSinceLastSummary = 0, initialSummarizerDelayMs = 0 } = isSummariesDisabled(
+			this.summaryConfiguration,
+		)
+			? {}
+			: {
+					...this.summaryConfiguration,
+					initialSummarizerDelayMs:
+						// back-compat: initialSummarizerDelayMs was moved from ISummaryRuntimeOptions
+						//   to ISummaryConfiguration in 0.60.
+						this.runtimeOptions.summaryOptions.initialSummarizerDelayMs ??
+						this.summaryConfiguration.initialSummarizerDelayMs,
+				};
+		const module = await import(
```

Comment:
I'm a bit lost RE non-conditional loading of this chunk.
This will not help with anything, no?
I think we want to put into this chunk only pieces that can be avoided if isSummarizerClient === false.
I.e. load it after if (isSummarizerClient) line

Reply:
There are no recent bundle size numbers in this PR, but I'd expect that they would not change with the way code is structured right now. Would be great to confirm that.

Reply:
In this initializeBaseState(), we are calling this function only if we are loading it.

Reply:
initializeSummarizer is only called if we want to load summarizer.

Reply:
Should I move that if condition here?

Reply:
It would be clearer to move the condition here yes

Reply:
I wanted to echo what Vlad said - There is an opportunity to make this more efficient by loading certain modules depending on whether a client is the summarizer or not (isSummarizerClient check). Summarizer clients only need to load `Summarizer` whereas non-summarizer clients need `SummaryManager`.

Reply:
@agarwal-navin I am doing that only i.e loading summarizer module based on if the client is summarizer or not.

Reply:
done